### PR TITLE
Add working groups page

### PR DESCRIPTION
**Pull request by samertm**
Adds the working group page.

 - Adds tables `working_groups` and `working_group_members`.
 - Vendors a fork of https://github.com/moreta/vue-search-select, modified to be similar to chosen-js.

Will follow up with a pull request that adds mailing list syncing.

### CODE REVIEW

File: model/db.go [hunk 1]
```diff
@@ -76,6 +78,32 @@ CREATE TABLE IF NOT EXISTS merged_activist_attendance (
...
+`)
+
+	db.MustExec(`
+CREATE TABLE IF NOT EXISTS working_group_members (
+  working_group_id INTEGER NOT NULL,
+  activist_id INTEGER NOT NULL,
+  -- True if the activist is the point person of the working group.
+  -- There should be only one point person per working group, but we
+  -- don't restrict that on the backend.
+  point_person TINYINT NOT NULL DEFAULT '0',
```
Comment (cbailey20043) [inline model/db.go]: What was the reason for moving the point person field back to the working_group_members? I felt it was much easier to enforce the "single point person per group" constraint this way.

**Comment by samertm** [comment]
I felt like it made the code easier to reason about b/c it makes
accessing + modifying working group members more uniform. Also, when
point_person is part of the working groups it's less clear what the
relationship is between the point person and the rest of
working_group_members (should the point person always exist in
working_group_members, or should it never?)

On Wed, Sep 27, 2017, at 09:08 PM, cbailey20043 wrote:
> *@cbailey20043* commented on this pull request.


> 
> In model/db.go[1]:


> > +  id INTEGER PRIMARY KEY AUTO_INCREMENT, +  name VARCHAR(60) NOT
> >    NULL, +  type TINYINT(1) NOT NULL, +  group_email VARCHAR(100)
> >    NOT NULL, +  CONSTRAINT working_groups_name_ukey UNIQUE (name) +)
> >    +`) + + db.MustExec(` +CREATE TABLE IF NOT EXISTS
> >    working_group_members ( +  working_group_id INTEGER NOT NULL, +
> >    activist_id INTEGER NOT NULL, +  -- True if the activist is the
> >    point person of the working group. +  -- There should be only one
> >    point person per working group, but we +  -- don't restrict that
> >    on the backend. +  point_person TINYINT NOT NULL DEFAULT '0',
>> What was the reason for moving the point person field back to the
> working_group_members? I felt it was much easier to enforce the
> "single point person per group" constraint this way.> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].> 

Links:

  1. https://github.com/dxe/adb/pull/25#discussion_r141522525
  2. https://github.com/dxe/adb/pull/25#pullrequestreview-65747721

File: model/working_groups.go [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+	}
+	return workingGroup.ID, nil
+}
+
+func insertWorkingGroupMembers(tx *sqlx.Tx, workingGroup WorkingGroup) error {
+	if workingGroup.ID == 0 {
+		return errors.New("Invalid WorkingGroup ID. ID's must be greater than 0")
+	}
+	// First drop all working group members for the working group.
+	_, err := tx.Exec(`DELETE FROM working_group_members WHERE working_group_id = ?`, workingGroup.ID)
```
Comment (cbailey20043) [inline model/working_groups.go]: Good idea. Completely forgot about doing that myself

**Comment by samertm** [comment]
> The one general comment I have (and you should go ahead and ship regardless if you agree or not, since its not a major thing) is that it seems a tad confusing that the point person is not listed as a member.

Ah, that's a good point. I'll ship this and we'll keep that in mind to fix in the future, since I'm not sure how that would work right now.

> Finally, I would probably let Michael review as well before shipping, since there are a lot of changes, and he may catch something that I missed (especially in the front end).

That's probably a good idea, but I'm eager to merge this :P Michael, if you have any comments feel free to leave them and I'll address them in another commit.

**Pull request merged**